mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -2587,7 +2642,7 @@ vectorizeScalableVectorPrecondition(Operation *op,
                  isa<linalg::MatmulTransposeAOp>(op) ||
                  isa<linalg::DepthwiseConv1DNwcWcOp>(op) ||
                  isa<linalg::MatvecOp>(op) || isa<linalg::Mmt4DOp>(op) ||
-                 hasReductionIterator(linalgOp));
+                 isa<linalg::UnPackOp>(op) || hasReductionIterator(linalgOp));
```

Comment:
Why do we need the change? I think we already check it above, so it is a deadcode?